From 448e946e11eb415f1c3a5c3e0ea6fddd57c0a351 Mon Sep 17 00:00:00 2001
From: Sean Purcell <me@seanp.xyz>
Date: Thu, 30 Mar 2017 16:35:21 -0700
Subject: [PATCH 1/1] Add patch for GENERIC instead of dstSize_tooSmall bug

---
 squashfs-tools/zstd_wrapper.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/squashfs-tools/zstd_wrapper.c b/squashfs-tools/zstd_wrapper.c
index 152353f..452056b 100644
--- a/squashfs-tools/zstd_wrapper.c
+++ b/squashfs-tools/zstd_wrapper.c
@@ -210,7 +210,13 @@ static int zstd_compress(void *strm, void *dest, void *src, int size,
 
 	if(ZSTD_isError(res)) {
 		const int errcode = ZSTD_getErrorCode(res);
-		if(errcode == ZSTD_error_dstSize_tooSmall) {
+		if(errcode == ZSTD_error_dstSize_tooSmall ||
+				/* FIXME:
+				 * ZSTD_compress 1.1.4 sometimes returns
+				 * GENERIC instead of dstSize_tooSmall,
+				 * this condition can be removed once the fix
+				 * is in a released version of zstd */
+				errcode == ZSTD_error_GENERIC) {
 			/* Special code for not enough buffer space */
 			return 0;
 		} else {
-- 
2.8.3

