From aa9d2d3e2cf0ed1fa3b5a3fafe712e907469261d Mon Sep 17 00:00:00 2001
From: Phillip Lougher <phillip@squashfs.org.uk>
Date: Sun, 6 Jul 2014 03:56:10 +0100
Subject: [PATCH 007/120] actions: readlink doesn't 0 terminate the returned
 path

Signed-off-by: Phillip Lougher <phillip@squashfs.org.uk>
---
 squashfs-tools/action.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/squashfs-tools/action.c b/squashfs-tools/action.c
index 85573f3..aa7bf2c 100644
--- a/squashfs-tools/action.c
+++ b/squashfs-tools/action.c
@@ -2316,7 +2316,10 @@ static int contained_fn(struct atom *atom, struct action_data *action_data)
 		 * filesystem */
 		return 0;
 
-	while(s != '\0' && depth) {
+	/* readlink doesn't 0 terminate the returned path */
+	s[bytes] = '\0';
+
+	while(depth) {
 		while(*s == '/')
 			s ++;
 
-- 
2.8.3

