From 48cc24893f0c2e69a388c6c45678a076316a9e25 Mon Sep 17 00:00:00 2001
From: Phillip Lougher <phillip@squashfs.org.uk>
Date: Wed, 27 Aug 2014 05:50:21 +0100
Subject: [PATCH 072/120] mksquashfs: add -true_action option

Signed-off-by: Phillip Lougher <phillip@squashfs.org.uk>
---
 squashfs-tools/mksquashfs.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/squashfs-tools/mksquashfs.c b/squashfs-tools/mksquashfs.c
index 37d9203..a1104f8 100644
--- a/squashfs-tools/mksquashfs.c
+++ b/squashfs-tools/mksquashfs.c
@@ -5208,6 +5208,17 @@ int main(int argc, char *argv[])
 			if(res == 0)
 				exit(1);
 
+		} else if(strcmp(argv[i], "-true_action") == 0 ||
+				strcmp(argv[i], "-ta") ==0) {
+			if(++i == argc) {
+				ERROR("%s: %s missing action\n",
+					argv[0], argv[i - 1]);
+				exit(1);
+			}
+			res = parse_action(argv[i], ACTION_LOG_TRUE);
+			if(res == 0)
+				exit(1);
+
 		} else if(strcmp(argv[i], "-af") == 0) {
 			if(++i == argc) {
 				ERROR("%s: -af missing filename\n", argv[0]);
-- 
2.8.3

