From 9c9e4304acee8e8e3ed1e9c398f28920d3216084 Mon Sep 17 00:00:00 2001
From: Phillip Lougher <phillip@squashfs.org.uk>
Date: Wed, 27 Aug 2014 05:52:20 +0100
Subject: [PATCH 073/120] mksquashfs: add -false_action option

Signed-off-by: Phillip Lougher <phillip@squashfs.org.uk>
---
 squashfs-tools/mksquashfs.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/squashfs-tools/mksquashfs.c b/squashfs-tools/mksquashfs.c
index a1104f8..4ee45b1 100644
--- a/squashfs-tools/mksquashfs.c
+++ b/squashfs-tools/mksquashfs.c
@@ -5219,6 +5219,17 @@ int main(int argc, char *argv[])
 			if(res == 0)
 				exit(1);
 
+		} else if(strcmp(argv[i], "-false_action") == 0 ||
+				strcmp(argv[i], "-fa") ==0) {
+			if(++i == argc) {
+				ERROR("%s: %s missing action\n",
+					argv[0], argv[i - 1]);
+				exit(1);
+			}
+			res = parse_action(argv[i], ACTION_LOG_FALSE);
+			if(res == 0)
+				exit(1);
+
 		} else if(strcmp(argv[i], "-af") == 0) {
 			if(++i == argc) {
 				ERROR("%s: -af missing filename\n", argv[0]);
-- 
2.8.3

