From 84d9f63f4480009ff235c945df105160e1104dd1 Mon Sep 17 00:00:00 2001
From: Phillip Lougher <phillip@squashfs.org.uk>
Date: Sat, 30 Aug 2014 03:08:31 +0100
Subject: [PATCH 082/120] actions: fix those underscores

Signed-off-by: Phillip Lougher <phillip@squashfs.org.uk>
---
 squashfs-tools/action.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/squashfs-tools/action.h b/squashfs-tools/action.h
index 1ca2df1..e1be949 100644
--- a/squashfs-tools/action.h
+++ b/squashfs-tools/action.h
@@ -188,7 +188,7 @@ struct type_entry {
 
 /*
  * Action logging requested, specified by the various
- * -action, -true_action, -false_action and -verbose_action
+ * -action, -true-action, -false-action and -verbose-action
  * options
  */
 #define ACTION_LOG_NONE	0
-- 
2.8.3

