From 83e05169a448dc9fe6f281248b141f708a5c00e0 Mon Sep 17 00:00:00 2001
From: Phillip Lougher <phillip@squashfs.org.uk>
Date: Tue, 18 Jul 2017 03:58:04 +0100
Subject: [PATCH 114/120] Fix compile warnings introduced with the -quiet
 option

Signed-off-by: Phillip Lougher <phillip@squashfs.org.uk>
---
 squashfs-tools/mksquashfs.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/squashfs-tools/mksquashfs.c b/squashfs-tools/mksquashfs.c
index 4808ae9..d36de17 100644
--- a/squashfs-tools/mksquashfs.c
+++ b/squashfs-tools/mksquashfs.c
@@ -4279,7 +4279,8 @@ void initialise_threads(int readq, int fragq, int bwriteq, int fwriteq,
 
 	main_thread = pthread_self();
 
-	quiet || printf("Parallel mksquashfs: Using %d processor%s\n", processors,
+	if(!quiet)
+		printf("Parallel mksquashfs: Using %d processor%s\n", processors,
 			processors == 1 ? "" : "s");
 
 	/* Restore the signal mask for the main thread */
@@ -5845,8 +5846,10 @@ printOptions:
 		void *comp_data = compressor_dump_options(comp, block_size,
 			&size);
 
-		quiet || printf("Creating %d.%d filesystem on %s, block size %d.\n",
-			SQUASHFS_MAJOR, SQUASHFS_MINOR, argv[source + 1], block_size);
+		if(!quiet)
+			printf("Creating %d.%d filesystem on %s, block size %d.\n",
+				SQUASHFS_MAJOR, SQUASHFS_MINOR,
+				argv[source + 1], block_size);
 
 		/*
 		 * store any compressor specific options after the superblock,
-- 
2.8.3

