From 1a6ffc749a7d9eb9f6425cb851746108e36c9127 Mon Sep 17 00:00:00 2001
From: Phillip Lougher <phillip@squashfs.org.uk>
Date: Mon, 31 Jul 2017 20:43:27 +0100
Subject: [PATCH 118/120] pseudo.c: Fix pseudo format error message

Signed-off-by: Phillip Lougher <phillip@squashfs.org.uk>
---
 squashfs-tools/pseudo.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/squashfs-tools/pseudo.c b/squashfs-tools/pseudo.c
index a3aebb5..cb74cf6 100644
--- a/squashfs-tools/pseudo.c
+++ b/squashfs-tools/pseudo.c
@@ -496,7 +496,7 @@ int read_pseudo_def(char *def)
 	return TRUE;
 
 error:
-	ERROR("Pseudo definitions should be of format\n");
+	ERROR("Pseudo definitions should be of the format\n");
 	ERROR("\tfilename d mode uid gid\n");
 	ERROR("\tfilename m mode uid gid\n");
 	ERROR("\tfilename b mode uid gid major minor\n");
-- 
2.8.3

