diff -up squashfs4.3/squashfs-tools/mksquashfs.c.omv~ squashfs4.3/squashfs-tools/mksquashfs.c
--- squashfs4.3/squashfs-tools/mksquashfs.c.omv~	2018-11-13 22:42:06.325534301 +0100
+++ squashfs4.3/squashfs-tools/mksquashfs.c	2018-11-13 22:43:30.537148778 +0100
@@ -47,6 +47,7 @@
 #include <pthread.h>
 #include <regex.h>
 #include <sys/wait.h>
+#include <sys/sysmacros.h>
 #include <limits.h>
 #include <ctype.h>
 
@@ -990,8 +991,8 @@ int create_inode(squashfs_inode *i_no, s
 	}
 	else if(type == SQUASHFS_CHRDEV_TYPE || type == SQUASHFS_BLKDEV_TYPE) {
 		struct squashfs_dev_inode_header *dev = &inode_header.dev;
-		unsigned int major = major(buf->st_rdev);
-		unsigned int minor = minor(buf->st_rdev);
+		unsigned int major = gnu_dev_major(buf->st_rdev);
+		unsigned int minor = gnu_dev_minor(buf->st_rdev);
 
 		if(major > 0xfff) {
 			ERROR("Major %d out of range in device node %s, "
@@ -1014,8 +1015,8 @@ int create_inode(squashfs_inode *i_no, s
 	}
 	else if(type == SQUASHFS_LCHRDEV_TYPE || type == SQUASHFS_LBLKDEV_TYPE) {
 		struct squashfs_ldev_inode_header *dev = &inode_header.ldev;
-		unsigned int major = major(buf->st_rdev);
-		unsigned int minor = minor(buf->st_rdev);
+		unsigned int major = gnu_dev_major(buf->st_rdev);
+		unsigned int minor = gnu_dev_minor(buf->st_rdev);
 
 		if(major > 0xfff) {
 			ERROR("Major %d out of range in device node %s, "
@@ -3554,7 +3555,7 @@ void dir_scan2(struct dir_info *dir, str
 		buf.st_mode = pseudo_ent->dev->mode;
 		buf.st_uid = pseudo_ent->dev->uid;
 		buf.st_gid = pseudo_ent->dev->gid;
-		buf.st_rdev = makedev(pseudo_ent->dev->major,
+		buf.st_rdev = gnu_dev_makedev(pseudo_ent->dev->major,
 			pseudo_ent->dev->minor);
 		buf.st_mtime = time(NULL);
 		buf.st_ino = pseudo_ino ++;
diff -up squashfs4.3/squashfs-tools/unsquashfs.c.omv~ squashfs4.3/squashfs-tools/unsquashfs.c
--- squashfs4.3/squashfs-tools/unsquashfs.c.omv~	2018-11-13 22:43:40.034330616 +0100
+++ squashfs4.3/squashfs-tools/unsquashfs.c	2018-11-13 22:43:56.332642564 +0100
@@ -33,6 +33,7 @@
 #include "fnmatch_compat.h"
 
 #include <sys/sysinfo.h>
+#include <sys/sysmacros.h>
 #include <sys/types.h>
 #include <sys/time.h>
 #include <sys/resource.h>
@@ -1131,7 +1132,7 @@ int create_inode(char *pathname, struct
 					unlink(pathname);
 
 				if(mknod(pathname, chrdev ? S_IFCHR : S_IFBLK,
-						makedev((i->data >> 8) & 0xff,
+						gnu_dev_makedev((i->data >> 8) & 0xff,
 						i->data & 0xff)) == -1) {
 					ERROR("create_inode: failed to create "
 						"%s device %s, because %s\n",
